### DISCUSSION TITLE
add subview for layer doesn't work

### DISCUSSION THREAD

**Discussion by romaHerman** [airbnb/lottie-ios#2212] [General]
Hello! 

I want to follow this tutorial https://github.com/brianadvent/DynamicLottieContent on how to add image to the Lottie animation layer. 
I see that there no method add view for layerName anymore 
it's now `addSubview(subview, forLayerAt: layerPath)`

here is code I'm trying to use 

for now I just don't see any results. My image doesn't being added
```
 
        animationView.frame = view.frame// Adjust the frame as needed
        animationView.contentMode = .scaleAspectFit
        animationView.loopMode = .loop
        let path = AnimationKeypath(keypath: "Placeholder")
        let subview = AnimationSubview(frame: CGRect(x: 0, y: 0, width: 200, height: 200))
        subview.addSubview(logo)
        animationView.addSubview(subview, forLayerAt: path)
        view.addSubview(animationView)
        animationView.play()
```

but unfortunately this code doesn't work

**Comment by calda** [comment]
`addSubview(forLayerAt:)` is only supported by the main thread rendering engine. To use this method you'll need to configure your `LottieAnimationView` to alway the main thread rendering engine, via the `LottieConfiguration`.

**Reply by romaHerman** [reply]
Thanks @calda 

here is my updated code 
unfortunately I'm receiving same result 
```
        LottieConfiguration.shared.renderingEngine = .mainThread
        animationView.frame = view.frame// Adjust the frame as needed
        animationView.contentMode = .scaleAspectFit
        animationView.loopMode = .loop
        let path = AnimationKeypath(keypath: "Placeholder")
        let subview = AnimationSubview(frame: CGRect(x: 0, y: 0, width: 200, height: 200))
        subview.addSubview(logo)
        animationView.addSubview(subview, forLayerAt: path)
        view.addSubview(animationView)
        animationView.play()
 ```

**Reply by calda** [reply]
Try passing in `configuration: LottieConfiguration(renderingEngine: .mainThread)` to `LottieAnimationView.init`.

Otherwise you need to update `LottieConfiguration.shared.renderingEngine` before calling `LottieAnimationView.init` (not after).

**Reply by romaHerman** [reply]
Thanks! it did the trick 
`subview.layer.contents = myImage.image?.cgImage` 
just one more adjustment I had to make instead of adding image as a subview and everything started to work!
